cmd/lifecycle-oracle/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+	"k8s.io/test-infra/prow/pjutil"
+	pprofutil "k8s.io/test-infra/prow/pjutil/pprof"
+	"k8s.io/test-infra/prow/version"
+)
+
+type options struct {
+	port                   int
+	instrumentationOptions prowflagutil.InstrumentationOptions
+	dataFile               string
+}
+
+func (o *options) validate() error {
+	if o.port == 0 {
+		return errors.New("--port is required")
+	}
+	if o.dataFile == "" {
+		return errors.New("--data-file is required")
+	}
+	return nil
+}
+
+func bindOptions(fs *flag.FlagSet) *options {
+	o := options{}
+	o.instrumentationOptions.AddFlags(fs)
+	fs.IntVar(&o.port, "port", 0, "Port to serve admission webhooks on.")
+	fs.StringVar(&o.dataFile, "data-file", "", "Path to file containing lifecycle phase data to serve.")
+	return &o
+}
+
+func main() {
```

Comment:
What advantage does this provide over just serving the file through a static webserver like nginx?

Reply:
It's a seed that's intended to grow but yes in this state it is no better than static